test/integration-tests/commands/build.test.ts

```diff
@@ -0,0 +1,91 @@
+import { WorkspaceContext } from "../../../src/WorkspaceContext";
+import { Commands } from "../../../src/commands";
+import { makeDebugConfigurations } from "../../../src/debugger/launch";
+import { Workbench } from "../../../src/utilities/commands";
+import { continueSession, waitForDebugAdapterCommand } from "../../utilities/debug";
+
+suite("Build Commands", function () {
+    let folderContext: FolderContext;
+    let workspaceContext: WorkspaceContext;
+    const uri = testAssetUri("defaultPackage/Sources/PackageExe/main.swift");
+    const breakpoints = [
+        new vscode.SourceBreakpoint(new vscode.Location(uri, new vscode.Position(2, 0))),
+    ];
+
+    suiteSetup(async function () {
+        workspaceContext = await globalWorkspaceContextPromise;
+        await waitForNoRunningTasks();
+        folderContext = await folderContextPromise("defaultPackage");
+        await workspaceContext.focusFolder(folderContext);
+        await vscode.window.showTextDocument(uri);
+        makeDebugConfigurations(folderContext, undefined, true);
+    });
+
+    suiteTeardown(async () => {
+        await vscode.commands.executeCommand(Workbench.ACTION_CLOSEALLEDITORS);
+    });
+
+    test("Swift: Run Build", async () => {
+        // A breakpoint will have not effect on the Run command.
+        vscode.debug.addBreakpoints(breakpoints);
+
+        const result = await vscode.commands.executeCommand(Commands.RUN);
+        expect(result).to.be.true;
+
+        vscode.debug.removeBreakpoints(breakpoints);
+    });
+
+    test("Swift: Clean Build", async () => {
+        const buildPath = path.join(folderContext.folder.fsPath, ".build");
+        const beforeItemCount = fs.readdirSync(buildPath).length;
+
+        const result = await vscode.commands.executeCommand(Commands.CLEAN_BUILD);
+        expect(result).to.be.true;
+
+        const afterItemCount = fs.readdirSync(buildPath).length;
+        // This test will run in order after the Swift: Run Build test,
+        // where .build folder is going to be filled with built artifacts.
+        // After executing the clean command the build directory is guranteed to have less entry.
+        expect(afterItemCount).to.be.lessThan(beforeItemCount);
+    });
+
+    test("Swift: Debug Build @slow", async () => {
+        vscode.debug.addBreakpoints(breakpoints);
+        // Promise used to indicate we hit the break point.
+        // NB: "stopped" is the exact command when debuggee has stopped due to break point,
+        // but "stackTrace" is the deterministic sync point we will use to make sure we can execute continue
+        const bpPromise = waitForDebugAdapterCommand(
+            "Debug PackageExe (defaultPackage)",
+            "stackTrace",
+            workspaceContext
+        );
+
+        const result = vscode.commands.executeCommand(Commands.DEBUG);
+        expect(result).to.eventually.be.true;
+
+        await bpPromise.then(() => continueSession());
```

Comment:
if we're hitting breakpoints, crossing into smoke level, let's create a smoke suite and put that there

Reply:
still should split into separate suite. Is it actually slow or just trying to keep out of CI build?

Reply:
What are we trying to achieve to split this into a separate suite? Is it for dependency management?

Given the test run fast and this is a command that should probably always work, I am not sure if we want keep this out of CI.

Reply:
will leave this for until long term approach is decided